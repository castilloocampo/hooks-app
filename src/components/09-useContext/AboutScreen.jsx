import React from 'react';
import PropTypes from 'prop-types';


const AboutScreen = () => {
  return (
    <div>
      <h1>AboutScreen</h1>
      <hr />
    </div>
  );
};


AboutScreen.propTypes = {

};


export default AboutScreen;
