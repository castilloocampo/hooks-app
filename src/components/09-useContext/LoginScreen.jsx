import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  const handleLogin = () => setUser({
    id: new Date().getTime(),
    name: "Mario Castillo",
    email: "mariotl9752@gmail.com"
  });

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button
        className='btn btn-primary'
        onClick={handleLogin} >Login</button>
    </div>
  );
};

export default LoginScreen;
