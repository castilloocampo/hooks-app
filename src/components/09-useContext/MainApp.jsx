import React from 'react';
import PropTypes from 'prop-types';


const MainApp = () => {
  return (
    <div>
      <h1>MainApp</h1>
      <hr />
    </div>
  );
};


MainApp.propTypes = {

};


export default MainApp;
