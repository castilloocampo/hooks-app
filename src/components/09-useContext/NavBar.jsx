import React from 'react';
import { NavLink } from 'react-router-dom';



const NavBar = () => {
  {/* <nav>
    <ul>
      <li>
        <Link to={`/`}> Home </Link>
      </li>
      <li>
        <Link to={`/about`}> About </Link>
      </li>
      <li>
        <Link to={`/login`}> Login </Link>
      </li>
    </ul>
</nav> */}

  const isActive = ({ isActive }) => {
    return `inline-block py-2 px-4 no-underline ${isActive ? 'text-gray-100' : 'text-gray-400'}`
  }

  return (
    <nav className="relative flex flex-wrap items-center content-between py-3 px-4  text-white bg-gray-800">
      <NavLink to={`/`} className="inline-block pt-1 pb-1 mr-4 text-lg whitespace-no-wrap"> useContext </NavLink>

      <div className="flex-grow items-center" id="navbarNavAltMarkup">
        <div className="flex flex-wrap list-reset pl-0 mb-0">
          <NavLink
            to={`/`}
            className={isActive}
          > Home </NavLink>
          <NavLink
            to={`/about`}
            className={isActive}
          > About </NavLink>
          <NavLink
            to={`/login`}
            className={isActive}
          > Login </NavLink>
        </div>
      </div>
    </nav>
  );
};

export default NavBar;
