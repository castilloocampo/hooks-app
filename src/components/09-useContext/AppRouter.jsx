import React from 'react';
import PropTypes from 'prop-types';
import {
  BrowserRouter as Router,
  Routes,
  Route,
  Navigate,
} from "react-router-dom";

import './styles.css'

import NavBar from './NavBar';
import AboutScreen from './AboutScreen';
import LoginScreen from './LoginScreen';
import HomeScreen from './HomeScreen';


const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />

        <Routes>
          <Route path='about' element={<AboutScreen />} />
          <Route path='login' element={<LoginScreen />} />
          <Route path='/' element={<HomeScreen />} />
          <Route path='*' element={<Navigate to={`/`} />} />

        </Routes>
      </div>
    </Router>
  );
};


AppRouter.propTypes = {

};


export default AppRouter;
