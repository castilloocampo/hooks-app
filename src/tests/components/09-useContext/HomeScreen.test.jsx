import { cleanup, render } from "@testing-library/react";
import { mount, shallow } from "enzyme"
import HomeScreen from '../../../components/09-useContext/HomeScreen';
import { UserContext } from "../../../components/09-useContext/UserContext";

const user = {
  name: 'Mario Castillo',
  email: 'mario@email.com'
}

// describe('HomeScreen component enzyme', () => {
//   const wrapper = mount(
//     <UserContext.Provider value={{
//       user: user
//     }}>
//       <HomeScreen />
//     </UserContext.Provider>
//   );

//   afterEach(() => {
//     cleanup();
//   });

//   test('should show correctly enzyme', () => {
//     expect(wrapper).toMatchSnapshot();
//   });

// });

describe('HomeScreen component React Testing Library', () => {

  let { container } = render(
    <UserContext.Provider value={{
      user: user
    }}>
      <HomeScreen />
    </UserContext.Provider>
  );

  afterEach(() => {
    cleanup();
  });

  test('should show correctly', () => {
    expect(container).toMatchSnapshot();
  });
});