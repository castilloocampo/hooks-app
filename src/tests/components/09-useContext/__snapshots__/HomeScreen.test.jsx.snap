// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen component React Testing Library should show correctly 1`] = `
<div>
  <div>
    <h1>
      HomeScreen
    </h1>
    <hr />
    <pre>
      {
   "name": "Mario Castillo",
   "email": "mario@email.com"
}
    </pre>
  </div>
</div>
`;
